maintainers/patches

```diff
@@ -0,0 +1,88 @@
+#!/bin/zsh
+
+PROGRAM=$0
+
+success() {
+  echo "\e[1;32mDone:\e[m $1"
+  exit 0
+}
+
+fail() {
+  echo "\e[1;31mError:\e[m $1"
+  exit 1
+}
+
+help() {
+  cat <<EOF
+Usage: $PROGRAM {apply|save}
+
+  apply       Applies the patches to the Tock submodule.
+              This resets Tock and can always be called.
+
+  save        Saves the Tock submodule to the patches.
+              Should only be called after apply and when all changes have been
+              added to a commit. After saving, you can run ./setup.sh to return
+              to normal state. Otherwise you can continue editing Tock and
+              calling save.
+EOF
+  exit 0
+}
+
+commit() {
+  local message=$1
+  git add .
+  git commit -qm$message
+}
+
+get_root() {
+  git ls-tree HEAD:third_party | sed -n 's/^.* \([^ ]\+\)\ttock$/\1/p'
+}
+
+get_head() {
+  git rev-parse HEAD
+}
+
+apply() {
+  git submodule -q update third_party/tock
+  ( cd third_party/tock
+    git reset -q --hard
+    git clean -qfxd
+    rsync -a ../../boards .
+    commit '00-boards'
+    for file in ../../patches/tock/*; do
+      patch -sp1 < $file
+      commit "${${file#../../patches/tock/}%.patch}"
+    done
+  )
+  success 'Applied the patches to the Tock submodule.'
+}
+
+save() {
+  local root=$(get_root)
+  ( cd third_party/tock
+    [[ -z "$(git status -s)" ]] || fail 'The Tock submodule is not clean.'
+    rm ../../patches/tock/*.patch
+    for file in $(git format-patch $root); do
+      sed -n '/^diff/,$p' $file | head -n-3 > ../../patches/tock/${file#*-}
```

Comment:
That's the only line that is ZSH specific. I'm not sure what the input/output are for this substitution so I can't give you a POSIX equivalent.
The rest of the script should be portable between zsh and bash AFAICT.

Reply:
It's actually not zsh specific. It still works with bash (at least on linux). The problem with the other line was that it was chaining/nesting such constructs which bash doesn't support.